### PR TITLE
Fixed Item price implementation

### PR DESCRIPTION
**Pull request by janelleljt**
In this PR I made some small changes to the implementation of item price. Namely, 
1. Store price as a Double instead of a BigDecimal
2. Updated checking of price validity
3. Fix the implementation of equals

### CODE REVIEW

**Review by donljh — APPROVED** [review]
~Looks good for now!~

File: src/main/java/tracko/model/item/Price.java [hunk 1]
```diff
@@ -30,23 +30,31 @@ public Price(BigDecimal price) {
      * @param test The price to validate.
      * @boolean Representing if the input is a valid price.
```
Comment (donljh) [inline src/main/java/tracko/model/item/Price.java]: The JavaDoc here should be
`@return True if input is a valid price`
instead of
`@boolean ....`

**Review by donljh — CHANGES_REQUESTED** [review]

File: src/main/java/tracko/model/item/Price.java (hunk 1 above)
Comment (janelleljt) [reply src/main/java/tracko/model/item/Price.java]: Updated!

**Review by donljh — APPROVED** [review]

**Review by sohenze — APPROVED** [review]
Looks good!

**Review by donljh — APPROVED** [review]

**Review by melissaharijanto — APPROVED** [review]

**Pull request merged**